qlib/rl/order_execution/state.py

```diff
@@ -38,6 +39,18 @@ def _get_all_timestamps(
     return pd.DatetimeIndex(ret)
 
 
+def fill_missing_data(
+    original_data: np.ndarray,
+    total_time_list: List[pd.Timestamp],
+    found_time_list: List[pd.Timestamp],
+    fill_method: Callable = np.median,
+) -> np.ndarray:
+    assert len(original_data) == len(found_time_list)
+    tmp = dict(zip(found_time_list, original_data))
+    fill_val = fill_method(original_data)
+    return np.array([tmp.get(t, fill_val) for t in total_time_list])
+
+
 class SAOEStateAdapter:
```

Comment:
I suggest moving this adapter to `state_adapter.py` or `simulator_qlib.py`. If I'm using the simple simluator, there is no reason I will be interested in all these adapter-related logics.

Can be left as TODO. :)